metpy/calc/tools.py

```diff
@@ -286,47 +287,135 @@ def reduce_point_density(points, radius, priority=None):
 
 
 @exporter.export
-def log_interp(x, xp, fp, **kwargs):
-    r"""Interpolates data with logarithmic x-scale.
+def log_interp(x, xp, *args, **kwargs):
+    r"""Interpolates data with logarithmic x-scale over a specified axis.
 
     Interpolation on a logarithmic x-scale for interpolation values in pressure coordintates.
 
     Parameters
     ----------
     x : array-like
-        The x-coordinates of the interpolated values.
+        1-D array of desired interpolated values.
 
     xp : array-like
         The x-coordinates of the data points.
 
-    fp : array-like
-        The y-coordinates of the data points, same length as xp.
+    args : array-like
+        The y-coordinates of the data points, same shape as xp.
+
+    axis : int, optional
+        The axis to interpolate over. Defaults to 0.
+
+    extrap: string, optional
+        Specify handling of interpolation points out of data bounds
+        'set_nan' : default, set value of points to 'nan'.
+        'clamp'   : set values to endpoint of data range.
+        'none'    : return ValueError if data is out of bounds.
 
     Returns
     -------
     array-like
-        The interpolated values, same shape as x.
+        List of interpolated values for each point.
 
-    Examples
-    --------
-    >>> x_log = np.array([1e3, 1e4, 1e5, 1e6])
-    >>> y_log = np.log(x_log) * 2 + 3
-    >>> x_interp = np.array([5e3, 5e4, 5e5])
-    >>> metpy.calc.log_interp(x_interp, x_log, y_log)
-    array([ 20.03438638,  24.63955657,  29.24472675])
+    Notes
+    -----
+    xp and args must be the same shape.
 
     """
-    sort_args = np.argsort(xp)
+    # Pull out keyword args
+    extrap = kwargs.pop('extrap', 'set_nan')
+    axis = kwargs.pop('axis', 0)
 
+    # Remove units from x coords
     if hasattr(x, 'units'):
         x = x.m
 
     if hasattr(xp, 'units'):
         xp = xp.m
 
-    interpolated_vals = np.interp(np.log(x), np.log(xp[sort_args]), fp[sort_args], **kwargs)
+    # Save number of dimensions in xp
+    ndim = xp.ndim
 
-    if hasattr(fp, 'units'):
-        interpolated_vals = interpolated_vals * fp.units
+    # Sort input data
+    sort_args = np.argsort(xp, axis=axis)
+    sort_x = np.argsort(x)
 
-    return interpolated_vals
+    # indicies for sorting
+    sorter = broadcast_indices(xp, sort_args, ndim, axis)
+
+    # Ensure pressure in increasing order
+    variables = []
+    variables = [arr[sorter] for arr in args]
+
+    # Make x broadcast with xp
+    if ndim == 1:
+        x_array = x[sort_x]
+    else:
+        x_array = x[sort_x]
+        for dim in range(ndim):
+            if dim < axis:
+                x_array = x_array[np.newaxis, ...]
+            elif dim > axis:
+                x_array = x_array[..., np.newaxis]
+
+    # Log x and xp
+    log_x_array = np.log(x_array)
+    log_xp = np.log(xp[sorter])
+
+    # Calculate value above interpolated value
+    minv = np.apply_along_axis(np.searchsorted, axis, xp[sorter], x[sort_x])
+    minv2 = np.copy(minv)
```

Comment:
I wanted to be sure to not overwrite mine since the values are used below to set the proper points to nan.